native/src/routes/SearchModal.tsx

```diff
@@ -37,51 +39,76 @@ const Wrapper = styled.View`
 `
 
 export type SearchModalProps = {
-  categories: CategoriesMapModel
+  allPossibleResults: Array<SearchResult>
   navigateTo: (routeInformation: RouteInformationType) => void
   theme: ThemeType
   languageCode: string
   cityCode: string
   closeModal: (query: string) => void
   t: TFunction<'search'>
   initialSearchText: string
+  loading: boolean
 }
 
 const SearchModal = ({
-  categories,
+  allPossibleResults,
   navigateTo,
   theme,
   languageCode,
   cityCode,
   closeModal,
   t,
   initialSearchText = '',
+  loading,
 }: SearchModalProps): ReactElement => {
   const [query, setQuery] = useState<string>(initialSearchText)
   const resourceCache = useResourceCache({ cityCode, languageCode })
 
-  const searchResults = useMemo(() => searchCategories(categories, query), [categories, query])
-
-  const onItemPress = (category: CategoryModel): void => {
-    const routeInformation: CategoriesRouteInformationType = {
-      route: CATEGORIES_ROUTE,
-      cityContentPath: category.path,
-      cityCode,
-      languageCode,
-    }
-    sendTrackingSignal({
-      signal: {
-        name: SEARCH_FINISHED_SIGNAL_NAME,
-        query,
-        url: urlFromRouteInformation(routeInformation),
+  const minisearch = useMemo(() => {
+    const search = new MiniSearch({
+      fields: ['title', 'content'],
+      storeFields: ['title', 'content', 'path', 'location', 'url', 'thumbnail'],
+      searchOptions: {
+        boost: { title: 2 },
+        fuzzy: true,
+        prefix: true,
       },
     })
-    navigateTo({
-      route: CATEGORIES_ROUTE,
-      cityCode,
-      languageCode,
-      cityContentPath: category.path,
-    })
+    search.addAll(allPossibleResults)
+    return search
+  }, [allPossibleResults])
+
+  // Minisearch doesn't add the returned storeFields (e.g. title or path) to its typing
+  const searchResults = minisearch.search(query) as unknown as SearchResult[]
```

Comment:
```suggestion
  // Minisearch doesn't add the returned storeFields (e.g. title or path) to its typing
I think the typing is correct, for me the return type is already `SearchResult[]` (and you are also using it without typecast in web).
```suggestion
  const searchResults = minisearch.search(query)
```
```

Reply:
Ah, yes, it returns the type `SearchResult` provided by `Minisearch` which is different from the one I defined (it doesn't have `title`, `content`, or `path`). When I remove the typecasting, the `List` yells at me because the function `renderItem` takes in our internal `SearchResults` as arguments, with title etc.

I am honestly somewhat confused about not having to typecast it in web but am happy to not need to?

Reply:
Weird stuff, seems like some weird naming problem... no idea about it tbh...